### PR TITLE
feat: add pointcloud realtime inference service

### PR DESCRIPTION
**Pull request by FrankGTY**
1. rewrite Dockerfile for service as container to deploy
    2. change voxelgenerator by using Point2VoxelGPU3d
    3. modify centerhead as center_head_single.py
    4. remove voxelsize keyword during build dense head in detector3d_template.py

### CODE REVIEW

File: openpcdet/config/cfgs.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import os
+
+DEFAULT_REDIS_HOST = "47.100.126.13"
```
Comment (ghost) [inline openpcdet/config/cfgs.py]: 不能这样写呀，写 172.17.0.1 或者 127.0.0.1 更好

File: openpcdet/config/cfgs.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+DEFAULT_REDIS_PASSWORD = "346aedb3f3d0"
```
Comment (wu-wenxiang) [inline openpcdet/config/cfgs.py:4]: 不能这样写呀，写 password 就好

File: openpcdet/tools/inference.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        return len(self.sample_file_list)
+
+    def __getitem__(self, index):
+        if self.ext == ".bin":
+            points = np.fromfile(self.sample_file_list[index]).reshape(-1, 4)
+            points = np.delete(points, np.where(points[:, 2] <= -1.7), axis=0)
+        elif self.ext == ".npy":
+            points = np.load(self.sample_file_list[index])
+        elif self.ext == "redis":
+            pcd = redis_client.rpop("127.0.0.1")
```
Comment (ghost) [inline openpcdet/tools/inference.py]: 这个应该读环境变量吧？或者 const 默认值

File: openpcdet/tools/cfgs/dataset_configs/my_dataset.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+DATASET: 'MyDataset'
+DATA_PATH: '/home/v2x/projects/datasets/zhongdian'
```
Comment (ghost) [inline openpcdet/tools/cfgs/dataset_configs/my_dataset.yaml]: 不要这样写……

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**